File: python-sdk/src/foremapi/models/shared/page.py
import dataclasses
from typing import Any,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from foremapi import utils

class PageTemplateEnum(str, Enum):
    CONTAINED = "contained"
    FULL_WITHIN_LAYOUT = "full_within_layout"
    NAV_BAR_INCLUDED = "nav_bar_included"
    JSON = "json"


@dataclass_json
@dataclasses.dataclass
class Page:
    r"""Page
    Representation of a page object
    """
    
    description: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('description') }})
    slug: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('slug') }})
    template: PageTemplateEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('template') }})
    title: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('title') }})
    body_json: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('body_json') }})
    body_markdown: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('body_markdown') }})
    is_top_level_path: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('is_top_level_path') }})
    social_image: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('social_image') }})
    

File: ts-sdk/src/sdk/models/shared/page.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum PageTemplateEnum {
    Contained = "contained",
    FullWithinLayout = "full_within_layout",
    NavBarIncluded = "nav_bar_included",
    Json = "json"
}


// Page
/** 
 * Representation of a page object
**/
export class Page extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=body_json" })
  bodyJson?: string;

  @SpeakeasyMetadata({ data: "json, name=body_markdown" })
  bodyMarkdown?: string;

  @SpeakeasyMetadata({ data: "json, name=description" })
  description: string;

  @SpeakeasyMetadata({ data: "json, name=is_top_level_path" })
  isTopLevelPath?: boolean;

  @SpeakeasyMetadata({ data: "json, name=slug" })
  slug: string;

  @SpeakeasyMetadata({ data: "json, name=social_image" })
  socialImage?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=template" })
  template: PageTemplateEnum;

  @SpeakeasyMetadata({ data: "json, name=title" })
  title: string;
}
